lib/models/builder.js

```diff
@@ -64,18 +72,31 @@ class Builder extends CoreObject {
       options = {
         tmpdir: tmpDir,
       };
-    } else {
-      broccoli = require('broccoli-builder');
-      if (this.systemTemp) {
-        console.warn('EMBER_CLI_SYSTEM_TEMP only works in combination with EMBER_CLI_BROCCOLI_2');
+
+      try {
+        this.builder = new broccoli.Builder(this.tree, options);
+        return;
+      } catch (e) {
+        // Catch here to trap InvalidNodeError. If this is thrown, it's because the node provided is not valid
+        // and likely uses the old .read/.rebuild API, so fallback to broccoli-builder that supports that API
```

Comment:
What other cases cause us to throw an `InvalidNodeError`?

Reply:
https://github.com/broccolijs/broccoli-node-info/blob/master/index.js#L49

So I've updated the catch to check the message string, bit hacky but whatever